### PR TITLE
Align dictionaries #DS-628

### PR DESCRIPTION
**Pull request by tomassychra**
# Align dictionary values [DS-628](https://jira.lmc.cz/browse/DS-628)

## ✅ What was done:
* dictionaries docs update
* React
   * added new value in `types/shared/dictionaries.ts`
   * `modalComposed.ts` `align` props
   * `ModalComposed/README.md` update
* Twig
   * Modal docs update
* Web
   * Modal docs update

### CODE REVIEW

File: packages/web-react/src/types/shared/dictionaries.ts [hunk 1]
```diff
@@ -58,3 +58,12 @@ export const ValidationStates = {
 
 export type ValidationStatesKeys = keyof typeof ValidationStates;
 export type ValidationStatesTypes<T> = (typeof ValidationStates)[ValidationStatesKeys] | T;
+
+export const Align = {
```
Comment (literat) [inline packages/web-react/src/types/shared/dictionaries.ts:62]: I am not sure about this naming. `align` seems to me too general and I can imagine situations where vertical and horizontal aligns should be used. In this case of the dictionary I would suggest using `HorizontalAlignment` instead. But we should have some concept about alignment.

**Review by literat — COMMENTED** [review]
Although it seems good to me I am not sure about the naming of the dictionary. Simple `align` or `alignment` in the case of horizontal alignment seems to me like a badly named dictionary. I do not care about the name of the prop but the constant/dictionary.

File: packages/web-react/src/types/shared/dictionaries.ts (hunk 1 above)
Comment (tomassychra) [reply packages/web-react/src/types/shared/dictionaries.ts]: After channel discussion, we agreed with `AlignmentX`, so i renamed to it.

File: docs/DICTIONARIES.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ This project uses `dictionaries` to unify props between different components.
 - [Size](#size)
 - [Color](#color)
 - [Validation](#validation)
+- [Align](#align)
```
Comment (literat) [inline docs/DICTIONARIES.md:15]: What about renaming it also here? The dictionary was renamed in code but not in the documentation.

Comment (tomassychra) [reply docs/DICTIONARIES.md]: done

File: packages/web-react/src/components/ModalComposed/useModalComposedStyleProps.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import classNames from 'classnames';
+import { AlignmentXDictionaryType } from '../../types';
```
Comment (crishpeen) [inline packages/web-react/src/components/ModalComposed/useModalComposedStyleProps.ts:2]: 🔡? :)

**Review by literat — APPROVED** [review]
LGTM

File: packages/web-react/src/components/ModalComposed/README.md [hunk 4]
```diff
@@ -110,12 +110,13 @@ The rest of the properties are created from the default `<div>` element. [Docs][
...
-| `children`         | `ReactNode`                 | -         | no       | Children node                 |
-| `align`            | `'left', 'center', 'right'` | `'right'` | no       | Children node                 |
-| `UNSAFE_className` | `string`                    | -         | no       | ModalFooter custom class name |
-| `UNSAFE_style`     | `CSSProperties`             | -         | no       | ModalFooter custom style      |
+| Name               | Type                                     | Default   | Required | Description                                                                  |
+| ------------------ | ---------------------------------------- | --------- | -------- | ---------------------------------------------------------------------------- |
+| `children`         | `ReactNode`                              | -         | no       | Children node                                                                |
+| `align`            | [Align dictionary][dictionary-alignment] | `'right'` | no       | [**DEPRECATED**][deprecated] in favor of `alignmentX`; ModalFooter alignment |
```
Comment (crishpeen) [inline packages/web-react/src/components/ModalComposed/README.md:116]:
```suggestion
| `align`            | [AlignmentX dictionary][dictionary-alignment] | `'right'` | no       | [**DEPRECATED**][deprecated] in favor of `alignmentX`; ModalFooter alignment |
```

Let's be concrete here. And update this everywhere. (just fulltext replace `[Align dictionary]` -> `[AlignmentX dictionary]`)

**Review by crishpeen — APPROVED** [review]
LGTM, just please rename the link in API tables.

File: packages/web-react/src/components/ModalComposed/README.md (hunk 4 above)
Comment (tomassychra) [reply packages/web-react/src/components/ModalComposed/README.md]: fixed

**Pull request merged**